### PR TITLE
add xxHash64 wrapper and convert most hashing

### LINKED ISSUE #1452: Add stronger hash or comparison to xDS APIs
Some xDS APIs (e.g. LDS, RDS and EDS) ignore updates that hash to the value of the existing config. This is to avoid churn and overhead when there is no actual update. However, we use `std::hash` today for this and no further checks, which is probably not strong enough. We should either keep the original value of the config proto around for direct comparison or use a stronger hash that will guarantee improbable collisions.

### PR DESCRIPTION
**Pull request by danielhochman**
followup to #1624 

left hashing in redis path alone for now. will figure out an internal strategy for migrating, possibly compiling in what std::hash is using so we don't have to worry about rehashing all of our cache data on deploy.

fixes #1452

### CODE REVIEW

File: source/common/common/hash.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+#include <string>
+
+#include "xxhash.h"
+
+namespace Envoy {
+
+class HashUtil {
+public:
+  /**
+   * Return 64-bit hash with seed of 0 from the xxHash algorithm.
```
Comment (mattklein123) [inline source/common/common/hash.h]: nit: Can you link to repo/docs so people can read about it if they want.

File: source/common/redis/conn_pool_impl.h [hunk 2]
```diff
@@ -154,7 +154,7 @@ class InstanceImpl : public Instance {
 
   struct LbContextImpl : public Upstream::LoadBalancerContext {
     LbContextImpl(const std::string& hash_key) : hash_key_(std::hash<std::string>()(hash_key)) {}
-
+    // TODO(danielhochman): convert to HashUtil::xxHash64 when we have a migration strategy
```
Comment (mattklein123) [inline source/common/redis/conn_pool_impl.h]: nit: period end of sentence

File: source/common/upstream/ring_hash_lb.cc [hunk 3]
```diff
@@ -107,6 +107,7 @@ void RingHashLoadBalancer::Ring::create(Runtime::Loader& runtime,
   for (const auto& host : hosts) {
     for (uint64_t i = 0; i < hashes_per_host; i++) {
       std::string hash_key(host->address()->asString() + "_" + std::to_string(i));
+      // TODO(danielhochman): convert to HashUtil::xxHash64 when we have a migration strategy
```
Comment (mattklein123) [inline source/common/upstream/ring_hash_lb.cc]: nit: period end of sentence

File: source/common/common/hash.h (hunk 1 above)
Comment (danielhochman) [reply source/common/common/hash.h]: done

**Pull request merged**